nextcord/polls.py

```diff
@@ -0,0 +1,425 @@
+            voted to the choice.
+        after: Optional[:class:`abc.Snowflake`]
+            For pagination, votes are sorted by member.
+        """
+        if limit is None:
+            limit = self.count
+
+        return AnswerVotersIterator(
+            message=self.poll.message, answer_id=self.id, after=after, limit=limit
+        )
+
+    def to_dict(self) -> PollAnswerCountPayload:
+        payload: PollAnswerCountPayload = {
+            "id": self.id,
+            "me_voted": self.me_voted,
+            "count": self.count,
+        }
+        return payload
+
+
+class PollResults:
+    """The result of the poll.
+
+    This contains the numbers of vote for each answer.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    is_finalized: :class:`bool`
+        Whether the votes has been precisely counted.
+    answer_counts: List[:class:`PollAnswerCount`]
+        The counts for each answer.
+    poll: :class:`Poll`
+        The poll this poll result represents.
+    """
+
+    __slots__ = (
+        "is_finalized",
+        "answer_counts",
+        "poll",
+    )
+
+    def __init__(self, data: PollResultsPayload, poll: Poll) -> None:
+        self.is_finalized: bool = data["is_finalized"]
+        self.answer_counts: List[PollAnswerCount] = [
+            PollAnswerCount(answer_count, poll) for answer_count in data["answer_counts"]
+        ]
+        self.poll: Poll = poll
+
+    def __repr__(self) -> str:
+        return (
+            "<PollResults "
+            f"is_finalized={self.is_finalized!r} "
+            f"answer_counts={self.answer_counts!r} "
+            ">"
+        )
+
+    def to_dict(self) -> PollResultsPayload:
+        payload: PollResultsPayload = {
+            "is_finalized": self.is_finalized,
+            "answer_counts": [answer_count.to_dict() for answer_count in self.answer_counts],
+        }
+        return payload
+
+
+class PollCreateRequest:
+    """A poll create request.
+
+    You must use this to create a poll.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    question: :class:`PollMedia`
+        The question of the poll. Currently only `text` is supported.
+    duration: Optional[:class:`int`]
+        The number of hours this poll should be open for. Defaults to ``24``. Max ``768`` (32 days).
+    allow_multiselect: Optional[:class:`bool`]
+        Whether voters can select more than one choice. Defaults to False.
+    answers: Optional[List[:class:`PollAnswer`]]
+        The answers to the poll.
+        If omitted then it must be added later through :func:`PollCreateRequest.add_answer`.
+    layout_type: Optional[:class:`PollLayoutType`]
+        The layout type of the poll. Defaults to :attr:`PollLayoutType.default`
+    """
+
+    __slots__ = (
+        "question",
+        "answers",
+        "duration",
+        "allow_multiselect",
+        "layout_type",
+    )
+
+    def __init__(
+        self,
+        question: PollMedia,
```

Comment:
IMO, `question` should accept both a `str` and a `PollMedia`, and converts it to a `PollMedia` if it isn't already a `PollMedia` object.